src/Persistence/Reflection/RuntimeReflectionProperty.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Persistence\Reflection;
+
+use Doctrine\Common\Proxy\Proxy as CommonProxy;
+use Doctrine\Persistence\Proxy;
+use ReflectionProperty;
+use ReturnTypeWillChange;
+
+/**
+ * PHP Runtime Reflection Property.
+ *
+ * Avoids triggering lazy loading if the provided object
+ * is a {@see \Doctrine\Persistence\Proxy}.
+ */
+class RuntimeReflectionProperty extends ReflectionProperty
+{
+    /** @var string */
+    private $key;
+
+    public function __construct(string $class, string $name)
+    {
+        parent::__construct($class, $name);
+        $this->key = $this->isPrivate() ? "\0" . $class . "\0" . $name : ($this->isProtected() ? "\0*\0" . $name : $name);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return mixed
+     */
+    #[ReturnTypeWillChange]
+    public function getValue($object = null)
+    {
+        if ($object === null) {
+            return parent::getValue($object);
+        }
+
+        return ((array) $object)[$this->key] ?? null;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param object|null $object
+     * @param mixed       $value
+     *
+     * @return void
+     */
+    #[ReturnTypeWillChange]
+    public function setValue($object, $value = null)
+    {
+        if (! ($object instanceof Proxy && ! $object->__isInitialized())) {
+            parent::setValue($object, $value);
+
+            return;
+        }
+
+        if ($object instanceof CommonProxy) {
+            $originalInitializer = $object->__getInitializer();
+            $object->__setInitializer(null);
+            parent::setValue($object, $value);
+            $object->__setInitializer($originalInitializer);
+
+            return;
+        }
+
+        $object->__setInitialized(true);
```

Comment:
Technically, it might not always exist, as the method is only defined in phpdoc in the interface for now, not yet enforced (which is precisely what allows to do that change in a minor version). So I think your removal of the check was a mistake

Reply:
I removed it because I thought that in practice the method will always be implemented, and removing the check would save a few CPU cycles. But I'm fine adding it back. PR updated.

Reply:
As the proxy generators are not in doctrine/persistence itself, it would be totally possible for doctrine/persistence to be updated without getting an updated proxy generator defining the new method.